src/serenity/config.hpp

```diff
@@ -11,140 +12,133 @@
 #include "serenity/serenity.hpp"
 
 #include "stout/option.hpp"
+#include "stout/result.hpp"
 
 namespace mesos {
 namespace serenity {
 
 /**
- * Global Serenity Config class which implements basic mechanism
- * to support specifying config parameters via string key map.
+ * Serenity Config class which implements basic mechanism
+ * to support specifying config parameters via string key map & sections.
  *
  * Check config_test.cpp to see example usage.
- *
- * TODO(skonefal): every getter should pack result in Try<T>.
  */
 class SerenityConfig {
  public:
   SerenityConfig() {}
 
   /**
-  * Variant type for storing multiple types of data in configuration.
-  */
+   * Variant type for storing multiple types of data in configuration.
+   */
   using CfgVariant = boost::variant<
-    bool, int64_t, uint64_t, double_t, std::string>;
+    bool, int64_t, double_t, std::string>;
 
   /**
    * Overlapping custom configuration options using recursive copy.
    */
   void applyConfig(const SerenityConfig& customCfg) {
-    this->recursiveCfgCopy(this, customCfg);
+    recursiveCfgCopy(this, customCfg);
   }
 
   /**
-   * Gets variant config value.
+   * Templated, safe getter for item in config.
    */
-  Option<SerenityConfig::CfgVariant> operator()(std::string key) const {
-    return getField(key);
-  }
+  template <typename T>
+  const Result<T> item(const std::string& key) const {
```

Comment:
"Nouns" are not best answer to question "what do this function do?".
It is better to use verbs for function names, and nouns for fields.
Also, it is best when names are self-explanatory and non-ambiguous. 

How about:
```
Result<T> getValue(const std::string& key);
T getValueOrDefault(const std::string& key, const T& default);
const Result<SerenityConifig&> getSection(const std::string& key);
const SerenityConifig& getSectionOrEmpty(const std::string& key);
```